public/js/post/post.js

```diff
@@ -124,6 +135,44 @@ var SpecialistPost = React.createClass({
         this.setState({selected: getCookie('specialist') === 'active'});
     },
 
+    isLevel400: function (course, level400Array) {
+        return notSpecialistCourse(course) && course.substring(3, 4) === '4' && level400Array.length < 3;
+    },
+
+    isLevel300: function (course, level300Array) {
+        return notSpecialistCourse(course) && course.substring(3, 4) >= '3' && level300Array.length < 3;
+    },
+
+    isLevelExtra: function (course, levelExtraArray) {
+        return notSpecialistCourse(course) && course.substring(3, 4) >= '3' && levelExtraArray.length < 4;
+    },
+
+    isInquiryCourse: function(course, inquiryArray) {
+        return CSCinq.indexOf(course) >= 0 && inquiryArray < 1;
+    },
+
+    getCourses: function () {
+        // [level400Courses, level300Courses, levelExtraCourses, inquiryCourse]
+        var courseArrays = [[], [], [], []];
```

Comment:
Use a loop (or something) to create this dyanmically, so that the length isn't hard-coded to 3.